streaming/src/test/scala/org/apache/spark/streaming/receiver/BlockGeneratorSuite.scala

```diff
@@ -184,9 +184,10 @@ class BlockGeneratorSuite extends SparkFunSuite with BeforeAndAfter {
     // Verify that the final data is present in the final generated block and
     // pushed before complete stop
     assert(blockGenerator.isStopped() === false) // generator has not stopped yet
-    clock.advance(blockIntervalMs)   // force block generation
```

Comment:
There is a race condition here. If `updateCurrentBuffer` runs before `blockIntervalTimer.stop(interruptTimer = false)`, this test will fail because RecurringTimer is blocking in this line: https://github.com/apache/spark/blob/69c9c177160e32a2fbc9b36ecc52156077fca6fc/streaming/src/main/scala/org/apache/spark/streaming/util/RecurringTimer.scala#L96